packages/app/src/specs/flaky-badge/FlakySpecSummary.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <div
+    class="border-t-4px min-w-200px w-full max-w-400px grid p-4 gap-4 grid-cols-1 justify-items-center"
+    :class="severity?.accentClass"
+    data-cy="flaky-spec-summary"
+  >
+    <SpecNameDisplay
+      :spec-file-name="specName"
+      :spec-file-extension="specExtension"
+    />
+    <div class="flex flex-row w-full text-size-14px justify-center items-center">
+      <component :is="severity?.icon" />
+      <span
+        class="font-medium ml-2"
+        :class="severity?.textClass"
+      >{{ severity?.label }}</span>
+      <span
+        class="ml-4"
+        data-cy="flaky-rate"
+      >{{ t('specPage.flaky.flakyRate', [flakyRate]) }}</span>
+    </div>
+
+    <div class="w-full grid text-gray-700 text-size-14px gap-2 grid-cols-2 justify-items-center">
+      <span data-cy="flaky-runs">{{ t('specPage.flaky.flakyRuns', { count: totalFlakyRuns, flakyRuns: totalFlakyRuns, totalRuns }) }}</span>
+      <span data-cy="last-flaky">{{ t('specPage.flaky.lastFlaky', { count: runsSinceLastFlake, runsSinceLastFlake }) }}</span>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+
+import { computed } from 'vue'
+import { useI18n } from '@cy/i18n'
+import SpecNameDisplay from '../SpecNameDisplay.vue'
+import LowRateIcon from '~icons/cy/rate-low_x16'
+import MediumRateIcon from '~icons/cy/rate-medium_x16'
+import HighRateIcon from '~icons/cy/rate-high_x16'
+
+const { t } = useI18n()
+
+const props = defineProps<{
+  specName: string
+  specExtension: string
+  severity: 'low' | 'medium' | 'high'
+  totalFlakyRuns: number
+  totalRuns: number
+  runsSinceLastFlake: number
+  dashboardUrl: string
+}>()
+
+const flakyRate = computed(() => {
+  if (props.totalFlakyRuns <= 0 || props.totalRuns <= 0) {
+    return 0
+  }
+
+  const rawRate = props.totalFlakyRuns / props.totalRuns * 100
+
+  // Only display 100 if rawRate is actually 100 (do not round to 100)
+  if (rawRate > 99 && rawRate < 100) {
+    return 99
+  }
+
+  return Math.ceil(rawRate)
+})
+
+const severity = computed(() => {
+  switch (props.severity) {
+    case 'low':
+      return {
+        accentClass: 'border-t-orange-400',
+        textClass: 'text-orange-400',
+        label: t('specPage.flaky.severityLow'),
+        icon: LowRateIcon,
+      }
+    case 'medium':
+      return {
+        accentClass: 'border-t-orange-500',
+        textClass: 'text-orange-500',
+        label: t('specPage.flaky.severityMedium'),
+        icon: MediumRateIcon,
+      }
+    case 'high':
+      return {
+        accentClass: 'border-t-orange-600',
+        textClass: 'text-orange-600',
+        label: t('specPage.flaky.severityHigh'),
+        icon: HighRateIcon,
+      }
+    default:
+      return undefined
+  }
+})
```

Comment:
```suggestion
const SEVERITIES = {
  low: {
    label: t('specPage.flaky.severity.low'),
    icon: LowRateIcon,
    textClass: 'text-gray-600',
    accentClass: 'bg-gray-100',
  },
  medium: {
    label: t('specPage.flaky.severity.medium'),
    icon: MediumRateIcon,
    textClass: 'text-gray-600',
    accentClass: 'bg-orange-100',
  },
  high: {
    label: t('specPage.flaky.severity.high'),
    icon: HighRateIcon,
    textClass: 'text-gray-600',
    accentClass: 'bg-red-100',
  },
}

const severity = computed(() => SEVERITIES[props.severity])
```

Optional, but I don't like when a component is littered with `?.` optional access when it isn't the components job to safeguard against nulls. The component is typed such that `severity` should not be null

Reply:
Regarding the `?.` usages: Normally I would agree. In this case the GQL schema is structured in a way that doesn't give us an typesafe enum for the severity value but rather just a string. Since ACI is reaching out to the cloud at runtime there's a chance that an accidental/unplanned change to the services could start sending back a value at some point in the future that's unknown to this version of the app, so I'm trying to structure this component in a way so the frontend handles it without blowing up.

I do like extracting the `switch` out to a `const` dictionary, will take a look at doing that

Reply:
In the case that the schema changes and it doesn't conform to `'low' | 'medium' | 'high'`, what does this component look like? Avoiding a runtime error is is best, but this component will display some wacky stuff. We could fallback to the low severity group if we don't match or define an undefined state, but I don't like mixing gql anomalies with a pure presentational component

Reply:
Yeah, falling back to the "low" state is probably a better way to handle this. Will update